products/cloudiot/api.yaml

```diff
@@ -192,3 +192,213 @@ objects:
                       output: true
                       description: |
                         The type of public key in the certificate.
+  - !ruby/object:Api::Resource
+    name: 'Device'
+    base_url: '{{registry}}/devices'
+    self_link: '{{registry}}/devices/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Google Cloud IoT Core device.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/iot/docs/'
+      api: 'https://cloud.google.com/iot/docs/reference/cloudiot/rest/'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: registry
+        input: true
+        url_param_only: true
+        required: true
+        description: |
+          The name of the device registry where this device should be created.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'id'
+        input: true
+        required: true
+        description: |
+          The unique identifier for the device. For example,
+          `Device0`.
+      - !ruby/object:Api::Type::String
+        name: 'name'
```

Comment:
Is this expected in the form documented, or will it take only the device name?

`projects/example-proj/locations/us-central1/registries/my-registry/devices/device0`
vs
`device0`

The long form of this name is tricky to build, so if the API requires that form explicitly we may want to expand that within the resource

Reply:
Oh this is excluded later, disregard this

Reply:
👍